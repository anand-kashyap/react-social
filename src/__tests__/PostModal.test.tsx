import React from 'react'
import { render, cleanup } from '@testing-library/react'
import PostModal from 'utils/modals/post/Modal';

const renderPModal = (opened = false, closed = () => { }) => {
  return render(
    <PostModal opened={opened} closed={closed} />
  );
};

describe('Post Modal Component', () => {
  afterEach(cleanup);

  it('should take a snapshot', () => {
    const { asFragment } = renderPModal();
    expect(asFragment()).toMatchSnapshot()
  });

  it('should autofocus textarea on each render', () => {
    const { getByTestId, rerender } = renderPModal(),
      txtArea = getByTestId('textarea');
    expect(txtArea).not.toHaveFocus();
    rerender(<PostModal opened={true} closed={() => { }} />);
    expect(txtArea).toHaveFocus();
  });

});