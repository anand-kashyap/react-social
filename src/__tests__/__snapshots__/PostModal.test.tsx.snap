// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Post Modal Component should take a snapshot 1`] = `
<DocumentFragment>
  <div
    class="modal null"
    id="modal01"
    tabindex="0"
  >
    <div
      class="modal-content"
    >
      <h5>
        Create Post
      </h5>
      <textarea
        data-testid="textarea"
        placeholder="Share something..."
      />
    </div>
    <div
      class="modal-footer"
    >
      <span>
        <strong>
          Return
        </strong>
         to send. 
        <strong>
          Shift + return
        </strong>
         to add new line
      </span>
      <p
        class="modal-close waves-effect waves-red btn-flat"
      >
        Cancel
      </p>
      <p
        class="modal-close waves-effect waves-light btn red"
      >
        Post
      </p>
    </div>
  </div>
</DocumentFragment>
`;
